src/Models/ModelInstantiator.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+namespace PDGA\DataObjects\Models;
+
+use PDGA\DataObjects\Attributes\Column;
+
+use ReflectionProperty;
+use ReflectionException;
+
+class ModelInstantiator
+{
+    /**
+     * Converts an associative array to a validated instance of a data object class.
+     *
+     * @param array  $arr   an array of values corresponding to the public properties of the data object.
+     * @param string $class Fully-qualified class name of the data object.
+     *
+     * @throws ReflectionException
+     * @return object
+     */
+    public function arrayToDataObject(
+        array  $arr,
+        string $class
+    ): object
+    {
+        $instance = new $class();
+
+        // TODO validate.
+
+        // Assign all public properties.
+        foreach ($this->dataObjectPropertyColumns($class) as $property => $column)
+        {
+            $instance->{$property} = $arr[$property] ?? null;
```

Comment:
These need to remain unset if the property doesn't exist on the array, they can't be set to null.  We need to be able to update records partially, e.g. just the `FirstName` of a member.  Let's say an update request comes in with this array:

```
['firstName' => 'Ben']
```

We map that to a `ModelInstantiatorTestObject`.  This code would set `LastName` etc. to null.  Then, down the line, the Repository code would set `LastName` etc. to NULL, which isn't right.  We want `LastName` etc. to be ignored.